src/components/home.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import './styles/home.css';
+import { NavLink } from 'react-router-dom';
+import { BsArrowRightCircle } from 'react-icons/bs';
+import { fetchStocks, select, search } from '../redux/homeSlice';
+
+const Home = () => {
+  const stocks = useSelector((state) => state.stocks.stocks);
+  const loading = useSelector((state) => state.stocks.loading);
+  const searchTerm = useSelector((state) => state.stocks.searchTerm);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    if (stocks.length === 0) dispatch(fetchStocks());
+  }, [dispatch, stocks.length]);
```

Comment:
Great way of ensuring you only fetch the data once in the lifecycle of this component! 🚀